### ISSUE TITLE
Oracle Linux 7.3 update 4 installation breaks on dependency resolution

### ISSUE THREAD

**Issue by adamancini** [docker/docker.github.io#4169]
File: [engine/installation/linux/docker-ee/oracle.md](https://docs.docker.com/engine/installation/linux/docker-ee/oracle/), CC @mstanleyjones 

the installation fails on package docker-ee-selinux conflicting with selinux-policy-targeted-3.13.1-166.0.2.el7.noarch

```
# yum install docker-ee
Loaded plugins: ulninfo
Resolving Dependencies
--> Running transaction check
---> Package docker-ee.x86_64 0:17.03.2.ee.5-1.el7 will be installed
--> Processing Dependency: selinux-policy >= 3.13.1-102.0.3.el7_3.15 for package: docker-ee-17.03.2.ee.5-1.el7.x86_64
--> Processing Dependency: docker-ee-selinux >= 17.03.2.ee.5-1.el7 for package: docker-ee-17.03.2.ee.5-1.el7.x86_64
--> Processing Dependency: libcgroup for package: docker-ee-17.03.2.ee.5-1.el7.x86_64
--> Processing Dependency: libltdl.so.7()(64bit) for package: docker-ee-17.03.2.ee.5-1.el7.x86_64
--> Processing Dependency: libseccomp.so.2()(64bit) for package: docker-ee-17.03.2.ee.5-1.el7.x86_64
--> Running transaction check
---> Package docker-ee-selinux.noarch 0:17.03.2.ee.5-1.el7 will be installed
--> Processing Dependency: selinux-policy-targeted >= 3.13.1-102.0.3.el7_3.15 for package: docker-ee-selinux-17.03.2.ee.5-1.el7.noarch
--> Processing Dependency: selinux-policy-base >= 3.13.1-102.0.3.el7_3.15 for package: docker-ee-selinux-17.03.2.ee.5-1.el7.noarch
--> Processing Dependency: policycoreutils-python for package: docker-ee-selinux-17.03.2.ee.5-1.el7.noarch
---> Package libcgroup.x86_64 0:0.41-13.el7 will be installed
---> Package libseccomp.x86_64 0:2.3.1-3.el7 will be installed
---> Package libtool-ltdl.x86_64 0:2.4.2-22.el7_3 will be installed
---> Package selinux-policy.noarch 0:3.13.1-102.0.2.el7_3.15 will be updated
---> Package selinux-policy.noarch 0:3.13.1-166.0.2.el7 will be an update
--> Running transaction check
---> Package policycoreutils-python.x86_64 0:2.5-17.1.0.1.el7 will be installed
--> Processing Dependency: policycoreutils = 2.5-17.1.0.1.el7 for package: policycoreutils-python-2.5-17.1.0.1.el7.x86_64
--> Processing Dependency: audit-libs-python >= 2.1.3-4 for package: policycoreutils-python-2.5-17.1.0.1.el7.x86_64
--> Processing Dependency: setools-libs >= 3.3.8-1 for package: policycoreutils-python-2.5-17.1.0.1.el7.x86_64
--> Processing Dependency: libsemanage-python >= 2.5-5 for package: policycoreutils-python-2.5-17.1.0.1.el7.x86_64
--> Processing Dependency: python-IPy for package: policycoreutils-python-2.5-17.1.0.1.el7.x86_64
--> Processing Dependency: libqpol.so.1(VERS_1.2)(64bit) for package: policycoreutils-python-2.5-17.1.0.1.el7.x86_64
--> Processing Dependency: libqpol.so.1(VERS_1.4)(64bit) for package: policycoreutils-python-2.5-17.1.0.1.el7.x86_64
--> Processing Dependency: libapol.so.4(VERS_4.0)(64bit) for package: policycoreutils-python-2.5-17.1.0.1.el7.x86_64
--> Processing Dependency: libapol.so.4()(64bit) for package: policycoreutils-python-2.5-17.1.0.1.el7.x86_64
--> Processing Dependency: libqpol.so.1()(64bit) for package: policycoreutils-python-2.5-17.1.0.1.el7.x86_64
---> Package selinux-policy-targeted.noarch 0:3.13.1-102.0.2.el7_3.15 will be updated
---> Package selinux-policy-targeted.noarch 0:3.13.1-166.0.2.el7 will be an update
--> Running transaction check
---> Package audit-libs-python.x86_64 0:2.7.6-3.el7 will be installed
--> Processing Dependency: audit-libs(x86-64) = 2.7.6-3.el7 for package: audit-libs-python-2.7.6-3.el7.x86_64
---> Package libsemanage-python.x86_64 0:2.5-8.el7 will be installed
--> Processing Dependency: libsemanage = 2.5-8.el7 for package: libsemanage-python-2.5-8.el7.x86_64
---> Package policycoreutils.x86_64 0:2.5-11.0.1.el7_3 will be updated
---> Package policycoreutils.x86_64 0:2.5-17.1.0.1.el7 will be an update
---> Package python-IPy.noarch 0:0.75-6.el7 will be installed
---> Package setools-libs.x86_64 0:3.3.8-1.1.el7 will be installed
--> Running transaction check
---> Package audit-libs.x86_64 0:2.6.5-3.el7_3.1 will be updated
---> Package audit-libs.x86_64 0:2.7.6-3.el7 will be an update
---> Package libsemanage.x86_64 0:2.5-5.1.el7_3 will be updated
---> Package libsemanage.x86_64 0:2.5-8.el7 will be an update
--> Processing Conflict: docker-ee-selinux-17.03.2.ee.5-1.el7.noarch conflicts docker-engine-selinux
--> Finished Dependency Resolution
Error: docker-ee-selinux conflicts with selinux-policy-targeted-3.13.1-166.0.2.el7.noarch
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

### workaround 
Install the version of selinux-policy-targeted before the conflicting selinux-policy-targeted-3.13.1-166.0.2.el7.noarch: 

`yum install selinux-policy-targeted-3.13.1-102.0.3.el7_3.16 `

After that docker-ee installs fine. 

Note that if you have a machine running OL7.4, you need to downgrade some selinux-policy-targeted dependencies first before installing selinux-policy-targeted-3.13.1-102.0.3.el7_3.16: 

`yum downgrade firewalld-0.4.3.2-8.1.el7_3.3 python-firewall-0.4.3.2-8.1.el7_3.3 firewalld-filesystem-0.4.3.2-8.1.el7_3.3 `

**Comment by andrewhsu** [comment]
Hmm...I'll have a look at this.

/cc @cpuguy83

**Comment by cpuguy83** [comment]
ping @Djelibeybi

**Comment by Djelibeybi** [comment]
We obsolete `docker-engine-selinux` with the latest `selinux-policy-targeted` on Oracle Linux 7.4. Because your `docker-ee-selinux` appears to conflict or obsolete `docker-engine-selinux`, you're ending up with unresolvable dependencies.

You may have to adjust the conflicts/obsoletes for `docker-ee-selinux` on Oracle Linux to resolve this. Or, determine if you even need `docker-ee-selinux` any more.

**Comment by Djelibeybi** [comment]
Also, can you please change the issue title from "OEL 7.3 update 4" to "Oracle Linux 7 Update 4". We haven't shipped a product named OEL for over 8 years now, the last version was 5.4. We'd prefer if you used Oracle Linux whenever possible, but if it must be abbreviated, please use OL instead. :)

**Comment by mstanleyjones** [comment]
While I appreciate the branding implications, I don't think we can expect Docker users to follow Oracle branding guidelines when filing Docker bugs. 👼

**Comment by Djelibeybi** [comment]
I don't expect them to either, which is why I'm taking the opportunity to correct the name of the product I manage. 😊

**Comment by mstanleyjones** [comment]
Closed via #4254
